bench/clblast/clblast_benchmark_level1.cpp

```diff
@@ -0,0 +1,281 @@
+  using ScalarT = ElemT;
+  size_t size = params;
+
+  ScalarT vr;
+  std::vector<ScalarT> buf1_host = benchmark<>::const_data<ScalarT>(size);
+  std::vector<ScalarT> buf2_host = benchmark<>::const_data<ScalarT>(size);
+
+  MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf1(*ex, buf1_host.data(), size);
+  MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf2(*ex, buf2_host.data(), size);
+  MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufr(*ex, &vr, 1);
+
+  Event event;
+  benchmark<>::flops_units_t flops =
+      benchmark<>::measure(reps, size * 2, [&]() {
+        clblast::Dot<ScalarT>(size, bufr.dev(), 0, buf1.dev(), 0, 1, buf2.dev(),
+                              0, 1, (*ex)._queue(), &event._cl());
+        event.wait();
+        event.release();
+      });
+
+  return flops;
+}
+
+BENCHMARK(iamax, clblast_level_1) {
+  using ScalarT = ElemT;
+  size_t size = params;
+
+  int vi;
+  auto buf1_host = benchmark<>::const_data<ScalarT>(size);
+
+  MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf1(*ex, buf1_host.data(), size);
+  MemBuffer<int, CL_MEM_READ_ONLY> buf_i(*ex, &vi, 1);
+
+  Event event;
+  benchmark<>::flops_units_t flops =
+      benchmark<>::measure(reps, size * 2, [&]() {
+        clblast::Amax<ScalarT>(size, buf_i.dev(), 0, buf1.dev(), 0, 1,
+                               (*ex)._queue(), &event._cl());
+        event.wait();
+        event.release();
+      });
+
+  return flops;
+}
+
+// not supported at current release yet
+BENCHMARK(iamin, clblast_level_1) {
+  using ScalarT = ElemT;
+  size_t size = params;
+
+  int vi;
+  auto buf1_host = benchmark<>::random_data<ScalarT>(size);
+
+  MemBuffer<ScalarT> buf1(*ex, buf1_host.data(), size);
+  MemBuffer<int> buf_i(*ex, &vi, 1);
+
+  Event event;
+  benchmark<>::flops_units_t flops =
+      benchmark<>::measure(reps, size * 2, [&]() {
+        clblast::Amin<ScalarT>(size, buf_i.dev(), 0, buf1.dev(), 0, 1,
+                               (*ex)._queue(), &event._cl());
+        event.wait();
+        event.release();
+      });
+
+  return flops;
+}
+
+BENCHMARK(scal2op, clblast_level_1) {
+  using ScalarT = ElemT;
+  size_t size = params;
+
+  ScalarT alpha(2.4367453465);
+  auto buf1_host = benchmark<>::random_data<ScalarT>(size);
+  auto buf2_host = benchmark<>::random_data<ScalarT>(size);
+
+  MemBuffer<ScalarT> buf1(*ex, buf1_host.data(), size);
+  MemBuffer<ScalarT> buf2(*ex, buf2_host.data(), size);
+
+  Event event1, event2;
+  benchmark<>::flops_units_t flops =
+      benchmark<>::measure(reps, size * 2, [&]() {
+        clblast::Scal<ScalarT>(size, alpha, buf1.dev(), 0, 1, (*ex)._queue(),
+                               &event1._cl());
+        clblast::Scal<ScalarT>(size, alpha, buf2.dev(), 0, 1, (*ex)._queue(),
+                               &event2._cl());
+        Event::wait({event1, event2});
+        Event::release({event1, event2});
+      });
+
+  return flops;
+}
+
+BENCHMARK(scal3op, clblast_level_1) {
+  using ScalarT = ElemT;
+  size_t size = params;
+
+  ScalarT alpha(2.4367453465);
```

Comment:
If these values are random, please make a note. Otherwise create a constant explaining why those values are selected

Reply:
I've added a `random_scalar` method to the benchmarking framework, that should let us unify all the random scalars, and make it clear that they're random (in the same way as `random_data` does for vectors/matrices).